### PR TITLE
Refactor `outputFile` APi in async/await

### PR DESCRIPTION
**Pull request by SukkaW**
The PR is a part of the proposal https://github.com/jprichardson/node-fs-extra/issues/1020.

All test cases pass: https://github.com/SukkaW/node-fs-extra/actions/runs/6599334843.

### CODE REVIEW

File: lib/output-file/index.js [hunk 1]
```diff
@@ -1,36 +1,29 @@
...
-
+async function outputFile (file, data, encoding = 'utf-8') {
   const dir = path.dirname(file)
-  pathExists(dir, (err, itDoes) => {
-    if (err) return callback(err)
-    if (itDoes) return fs.writeFile(file, data, encoding, callback)
 
-    mkdir.mkdirs(dir, err => {
-      if (err) return callback(err)
+  const itDoes = await pathExists(dir)
```
Comment (RyanZim) [inline lib/output-file/index.js:12]: Let's rename this to something sensible like `dirExists`, while we're here :grinning:

File: lib/output-file/index.js [hunk 2]
```diff
@@ -1,36 +1,29 @@
...
 
-      fs.writeFile(file, data, encoding, callback)
-    })
-  })
+  if (!itDoes) {
```
Comment (RyanZim) [inline lib/output-file/index.js:14]: Alternately, just inline `await pathExists(dir)` in the `if` expression, to match the sync version.

**Review by RyanZim — COMMENTED** [review]
LGTM, just one stupid variable name from the old code

File: lib/output-file/index.js (hunk 2 above)
Comment (SukkaW) [reply lib/output-file/index.js]: Fixed in [`990ffd2` (#1024)](https://github.com/jprichardson/node-fs-extra/pull/1024/commits/990ffd25209b8e6b4e9dd2a8c07d387cbaafbb53)

**Review by RyanZim — APPROVED** [review]

**Pull request merged**